cli/src/pcluster/config/update_policy.py

```diff
@@ -153,8 +175,60 @@ def __eq__(self, other):
     >= (change.old_value if change.old_value is not None else DEFAULT_MAX_COUNT),
 )
 
+
+def fail_reason_queue_update_strategy(change, _):
+    reason = "All compute nodes must be stopped"
+    # QueueUpdateStrategy can override UpdatePolicy of parameters under SlurmQueues
+    if any(path.startswith("SlurmQueues[") for path in change.path):
+        reason += " or queue update strategy must be set"
+
+    return reason
+
+
+def condition_checker_queue_update_strategy(change, patch):
```

Comment:
We're adding many "queue_update_strategy" related functions. 

What about moving them into a new module or adding them as methods of a new `QueueUpdateStrategy` class? (eventually with static methods only).

In this way they will be grouped together and if we are going to change this logic we'll have a single class/module to search for.

Reply:
adding two methods for queue_update_strategy. I prefer to leave as it is for now